File: js/Distribute Coins in Binary Tree.js
var distributeCoins = function(root) {
    let moves = 0;

    function postorder(node) {
        if (!node) return 0;
        const leftBalance = postorder(node.left);
        const rightBalance = postorder(node.right);
        const balance = node.val + leftBalance + rightBalance - 1;
        moves += Math.abs(balance);
        return balance;
    }

    postorder(root);
    return moves;
};

File: python/Distribute Coins in Binary Tree.py
class Solution:
    def distributeCoins(self, root: Optional[TreeNode]) -> int:
        self.moves = 0
        self.postorder(root)
        return self.moves
    
    def postorder(self, node: Optional[TreeNode]) -> int:
        if not node:
            return 0
        left_balance = self.postorder(node.left)
        right_balance = self.postorder(node.right)
        balance = node.val + left_balance + right_balance - 1
        self.moves += abs(balance)
        return balance